statsmodels/multivariate/glm.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+
+"""General linear model
+
+author: Yichuan Liu
+"""
+from __future__ import print_function, division
+
+from statsmodels.base.model import Model
+import numpy as np
+from numpy.linalg import eigvals, inv, solve, matrix_rank, pinv, svd
+from scipy import stats
+import pandas as pd
+from statsmodels.iolib import summary2
+
+
+def fit_glm(x, y, method='svd'):
+    """
+    Solve general linear model y = x * params
+    where y is dependent variables, x is independent variables
+
+    No acture
+
+    Parameters
+    ----------
+    x : array-like, each column is a independent variable
+    y : array-like, each column is a dependent variable
+
+    Returns
+    -------
+    a tuple of matrices or values necessary for hypothesis testing
+
+    .. [1] https://support.sas.com/documentation/cdl/en/statug/63033/HTML/default/viewer.htm#statug_introreg_sect012.htm
+
+    """
+    nobs, k_endog = y.shape
+    nobs1, k_exog= x.shape
+    if nobs != nobs1:
+        raise ValueError('x(n=%d) and y(n=%d) should have the same number of '
+                         'rows!' % (nobs1, nobs))
+
+    # Calculate the matrices necessary for hypothesis testing
+    df_resid = nobs - k_exog
+    tolerance = 1e-8
+    if method == 'pinv':
+        # Regression coefficients matrix
+        params = pinv(x).dot(y)
+
+        # inverse of x'x
+        inv_cov = inv(x.T.dot(x))
```

Comment:
check OLS: I think this is just pinv(x).dot(pinv(x)), where we should keep pinv(x) from the previous line